test/cli/fixtures/expected/tap-outputs.js

```diff
@@ -107,30 +107,29 @@ Extra reporters found among package dependencies: npm-reporter`,
 Last test to run (hanging) has an async hold. Ensure all assert.async() callbacks are invoked and Promises resolve. You should also set a standard timeout via QUnit.config.testTimeout.`,
 	/* eslint-enable max-len */
 	"qunit unhandled-rejection.js":
-`TAP version 13
-not ok 1 Unhandled Rejections > test passes just fine, but has a rejected promise
+`not ok 1 global failure
   ---
-  message: global failure: Error: Error thrown in non-returned promise!
+  message: Error: outside of a test context
   severity: failed
-  actual  : undefined
-  expected: undefined
   stack: |
-    Error: Error thrown in non-returned promise!
-        at /qunit/test/cli/fixtures/unhandled-rejection.js:10:10
+    Error: outside of a test context
+        at /qunit/test/cli/fixtures/unhandled-rejection.js:17:18
+        at processModule (/qunit/qunit/qunit.js)
+        at Object.module$1 [as module] (/qunit/qunit/qunit.js)
+        at /qunit/test/cli/fixtures/unhandled-rejection.js:3:7
         at internal
   ...
-not ok 2 global failure
+Bail out! Error: outside of a test context
+TAP version 13
```

Comment:
The unhandled rejection in this test case happens during the loading of the test file, before `QUnit.start/QUnit.begin` and thus before the reporter is sees `runStart`. I considered moving the `TAP version 13` line to the TapReporter constructor but that felt wrong as I don't think instantiating a reporter should already do something.

Perhaps in a follow-up (or in the same PR?) we could improve on this by instead making it be printed on the first relevant event, which will usually still be runStart, but would be lazily brought forward if for some reason a different event happens first (e.g. "error").